File: src/test/java/com/methodpark/cmpe/code/srp/CsvReaderTest.java
package com.methodpark.cmpe.code.srp;

import java.net.URI;
import java.net.URISyntaxException;
import java.util.List;

import junit.framework.Assert;

import org.junit.Test;

public class CsvReaderTest
{

    @Test
    public void getHeader_ValidFile_CorrectHeaderCount()
    {
        CsvReader reader = createCsvReader();

        String[] header = reader.getHeader();
        Assert.assertNotNull(header);
        Assert.assertEquals(5, header.length);
        Assert.assertEquals("Record1", header[0]);
    }

    @Test
    public void getRecords_ValidFile_CorrectRecordCount()
    {
        CsvReader reader = createCsvReader();

        List<String[]> records = reader.getRecords();

        Assert.assertNotNull(records);
        Assert.assertEquals(3, records.size());

        String[] firstRecord = records.get(0);

        Assert.assertEquals(5, firstRecord.length);
        Assert.assertEquals("1", firstRecord[0]);
    }

    private CsvReader createCsvReader()
    {
        CsvReader reader = new CsvReader(getTestFile());
        return reader;
    }

    private URI getTestFile()
    {
        try
        {
            return CsvReaderTest.class.getResource("TestFile.csv").toURI();
        }
        catch (URISyntaxException e)
        {
            return null;
        }
    }

}

File: src/test/csharp/Srp/SrpTests/CsvReaderTest.cs
using System.IO;
using FluentAssertions;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Srp;

namespace SrpTests
{
    [TestClass]
    public class CsvReaderTest
    {
        [TestMethod]
        public void GetHeader_ValidFile_CorrectHeaderCount()
        {
            var reader = CreateCsvReader();
            var header = reader.Header;
            header.Should().NotBeNull();
            header.Length.Should().Be(5);
            header[0].Should().Be("Record1");
        }

        [TestMethod]
        public void GetRecords_ValidFile_CorrectRecordCount()
        {
            var reader = CreateCsvReader();

            var records = reader.Records;

            records.Should().NotBeNull();
            records.Count.Should().Be(3);

            var firstRecord = records[0];
            firstRecord.Length.Should().Be(5);
            firstRecord[0].Should().Be("1");
        }

        private CsvReader CreateCsvReader()
        {
            var reader = new CsvReader(GetTestFile());
            return reader;
        }

        private string GetTestFile()
        {
            var stream = GetType().Assembly.GetManifestResourceStream("SrpTests.Resources.TestFile.csv");
            var csvFile = Path.GetTempFileName();
            File.WriteAllText(csvFile, new StreamReader(stream).ReadToEnd());
            return csvFile;
        }
    }
}